neonvm/controllers/vm_controller.go

```diff
@@ -800,6 +800,11 @@ func (r *VMReconciler) doReconcile(ctx context.Context, vm *vmv1.VirtualMachine)
 		// do nothing
 	}
 
+	if vm.Status.Phase == vmv1.VmRunning && vm.Spec.TargetRevision != nil {
+		rev := vm.Spec.TargetRevision.WithTime(time.Now())
+		vm.Status.CurrentRevision = &rev
+	}
```

Comment:
Just double-checking I understand correctly: This means that spec -> status propagation is only performed once the VM reaches the Running phase, right?

I think _in practice_ this should be fine, but it's worth considering what cases this doesn't catch...

Reply:
> Just double-checking I understand correctly: This means that spec -> status propagation is only performed once the VM reaches the Running phase, right?

Yes

> I think in practice this should be fine, but it's worth considering what cases this doesn't catch...

Well, the main case where we want to propagate the revision is when the VM transitions Scaling -> Running. Or when it didn't need scaling, and keeps being Running. I don't know of any other cases where it is need, actually.